### ISSUE TITLE
Vertical positioning/stacking issue on refresh with Safari 10.1

### ISSUE THREAD

**Issue by makingthingswork** [desandro/masonry#955]
-

**Comment by nuclearsugar** [comment]
I can confirm this issue when using Safari 10.1.

Visually it appears that padding is added to the bottom of each Masonry element. And this extra padding is the same height as it’s parent Masonry element. 

Interestingly, the Masonry elements return to expected behavior after the container changes size.

![screenshot1](https://cloud.githubusercontent.com/assets/8218211/25570389/d3779c5a-2df2-11e7-8bfc-0e079a4b4a34.png)

![screenshot2](https://cloud.githubusercontent.com/assets/8218211/25570390/d37a37f8-2df2-11e7-9612-75a2424187cc.png)

**Comment by makingthingswork** [comment]
Yes, thats exactly the issue.

I was stuck on this for ages and I looked in all the wrong areas for the problem. It was clear it was some sort loading issue but I couldn't identify the source of the problem.

I discovered that if I removed my custom fonts and used system fonts (or even Google web fonts) the issue is gone. I now trigger masonry after I have checked that the fonts declared using @font-face have actually loaded, I use Font Face Observer. 


```
var $grid = $('.grid');
		
//Check fonts have loaded then proceed
Promise.all([BrandonGrotesqueBold.load(), BrandonGrotesqueRegular.load()]).then(function () {

	$grid.imagesLoaded( function() {

		$grid.masonry({
			itemSelector: '.item',
			columnWidth: '.grid-sizer',
			gutter: '.gutter-sizer',
			percentPosition: true,
			stamp: '.stamp',
			//initLayout: false
		});
			  
	});

});

// add event listener for initial layout
$grid.on( 'layoutComplete', function( event, items ) {
	$('.stamp.page1,.item.page1').addClass('reveal');
});
```

This is not something I'd prefer to do but unfortunately I'm kind of locked into a specific font but if I can locate a suitable alternative or a very close match on Google web fonts (or other web font service), I'll hopefully be able to switch.

The font files are big and even if I convert them they are still comparatively way bigger to say the files loaded via Google web font services. My knowledge of fonts means I'm not really informed enough to know how to significantly reduce the file size, if thats even possible.

When I had identified the source of my problem I just tried to remove the post as I knew I hadn't followed the guidelines. It had been a very long time since I last posted and no such restrictions existed back then but I'd hit a wall in terms of ideas.

![screen shot 2017-04-27 at 19 24 14](https://cloud.githubusercontent.com/assets/998619/25573075/b1016b1e-2e39-11e7-9bd3-76b836167767.png)
![screen shot 2017-04-27 at 19 24 22](https://cloud.githubusercontent.com/assets/998619/25573078/b6c51b36-2e39-11e7-8176-fa86d12ad683.png)

**Comment by makingthingswork** [comment]
@nuclearsugar I notice that the http://masonry.desandro.com is loading fonts via @font-face as well. Strange that this issue manifests itself in this way.

**Comment by desandro** [comment]
@makingthingswork Thanks so much for looking into this issue.

Looks like [Font Face Observer](https://fontfaceobserver.com/) is a suitable solution

``` js
// initialize masonry
var msnry = new Masonry(...)

// trigger layout after font is loaded
var font = new FontFaceObserver('Texta');
font.load().then(function () {
  msnry.layout();
});
```

**Comment by makingthingswork** [comment]
@desandro No problem, been using Masonry for many years and had never encountered anything like  this before so I was really confused for a while.

Of course I realise its not an issue with Masonry - it very much looks like its directly related to how Safari is handling fonts being loaded. The size of the Typekit kit for my current project means I'll still be triggering Masonry after the Font Face Observer check.

I noticed almost the exact same issue occurs with jquery-backstretch when the container with the background image contains text that is styled using the custom font. If I don't check the fonts are loaded first before initialising backstretch the target container height is significantly bigger than it should be.

**Comment by nuclearsugar** [comment]
I’ve been banging my head against this issue and haven’t been able to successfully implement the solution you suggested. Am I missing something?

I tried to make a JSfiddle but couldn’t get the font issue to reproduce in the sandbox, so I created a barebones test on my own server - http://home.jasonfletcher.info/all/masonry_fix/


```
// initialize the Masonry plugin
var msnry = new Masonry( '#masonry_container', {
	itemSelector: '.info_box'
});

// reinitialize Masonry plugin after Google Font is loaded
var font = new FontFaceObserver('Raleway');
font.load().then(function () {
	msnry.layout(),
	alert("FontFaceObserver: Raleway - Google Font Loaded");
});
```

**Comment by makingthingswork** [comment]
@nuclearsugar You are loading multiple weights of the same typeface from Google Fonts and for the purposes of the FontFaceObservor you need to treat them as different fonts. Immediately it didn't seem logical to me but when I took a moment to think about it I realised its loading a file for each weight.

The check of multiple fonts is detailed here - https://github.com/bramstein/fontfaceobserver - towards the end of the How To Use section.

I initially encountered this same issue and modified my approach:

```
//Declare font variables
var BrandonGrotesqueBold = new FontFaceObserver('brandon-grotesque', { weight: 700 });
var BrandonGrotesqueRegular = new FontFaceObserver('brandon-grotesque', { weight: 400 });

//Check fonts have loaded then proceed
Promise.all([BrandonGrotesqueBold.load(), BrandonGrotesqueRegular.load()]).then(function () {

     //Do things here such as masonry
     msnry.layout();

});

```

**Comment by nuclearsugar** [comment]
@makingthingswork Thanks for the tip. Unfortunately I’m still seeing the same issue in my barebones example. I tried both the individual font name and font family name, but then saw that FontFaceObserver wants the font family name. I've also tried calling the msnry.layout() differently, to no effect. Any ideas?

Updated barebones test - http://home.jasonfletcher.info/all/masonry_fix/

```
// initialize the Masonry plugin
var msnry = new Masonry( '#masonry_container', {
	itemSelector: '.info_box'
});

//Declare font variables
var Raleway300 = new FontFaceObserver('Raleway', { weight: 300 });
var Raleway400 = new FontFaceObserver('Raleway', { weight: 400 });
var Raleway500 = new FontFaceObserver('Raleway', { weight: 500 });
var Raleway600 = new FontFaceObserver('Raleway', { weight: 600 });
var Raleway700 = new FontFaceObserver('Raleway', { weight: 700 });

//Check fonts have loaded then proceed
Promise.all([Raleway300.load(), Raleway400.load(), Raleway500.load(), Raleway600.load(), Raleway700.load()]).then(function () {
	msnry.layout(),
	alert("FontFaceObserver: Raleway - all font weights loaded");
});
```

**Comment by makingthingswork** [comment]
@nuclearsugar I noticed this thread over at FontFaceObserver that seems to allude to an issue with the script combined with Google Fonts. I'm using Typekit but I had a look at your test and everything looks as it should be.
https://github.com/bramstein/fontfaceobserver/issues/68

Legacy versions are available for download.
https://github.com/bramstein/fontfaceobserver/releases?after=v2.0.0

**Comment by nuclearsugar** [comment]
Ah interesting, I hadn't even considered that it might be a bug within FontFaceObserver. I tried downgrading to FontFaceObserver 1.7.1 and am still seeing the same issue. Alas!

For now I've implemented a quick hack. Since my particular issue to specific to Safari 10.1 and Google Fonts, I've found some CSS code to alter the font only for Safari 10.1 and rely on a system font instead. Using a system font doesn't trigger the Masonry extra padding issue outlined above. It's not an ideal solution but I'm out of time.

```
/* This CSS targets only Safari 10.1+ */
/* http://stackoverflow.com/questions/16348489/is-there-a-css-hack-for-safari-only-not-chrome */
@media not all and (min-resolution:.001dpcm) { @media {
html, body {font-family: 'Helvetica Neue', Arial, sans-serif !important;}
}}
```
